### ISSUE TITLE
loading..

### ISSUE THREAD

**Issue by mirrormonark** [KristjanESPERANTO/MMM-OneCallWeather#2]
My config:

```js
		{
			module: "onecall/loader",
			disabled: false,
			config: {
				lat: "XXXX",                               // your location latitude,
				lon: "XXXX",                               // your location longitude,
				appid: "XXXXXXXXXXXXXX",                             // your Openweathermap appid
				appid2: "",                            // optional
				backup: "",                            // optional backup appid
				dayUpdateInterval: 10 * 60 * 1000,     // every 10 minutes
				nightUpdateInterval: 15 * 60 * 1000,   // every 15 minutes
			}
		},
		{	
			module: "MMM-OneCallWeather",
			position: "top_right",	// This can be any of the regions.
						        	// Best results in bottom_bar region due to horizontal default layout and icon sizes.
			pages: {"daily":"top_right"},
			disabled: false,
			config: {
				// See 'Configuration options' for more information.
				latitude: "XXXXX", //Latitude from https://www.latlong.net/
				longitude: "XXXXX", //Longitude from https://www.latlong.net/ - is West + is East of London
				apikey: "XXXXXXXXXXXXXXXXXXX" //openweathermap.org API key.
			}
		},
```
I'm using the modules "page-indicator" and "page-selector" in my mirror

**Comment by mirrormonark** [comment]
@KristjanESPERANTO

**Comment by KristjanESPERANTO** [comment]
Where did you get that `module: "onecall/loader"`? I have never seen such a module.

Try this:
```js
		{	
			module: "MMM-OneCallWeather",
			position: "top_right",	// This can be any of the regions.
						        	// Best results in bottom_bar region due to horizontal default layout and icon sizes.
			pages: {"daily":"top_right"},
			disabled: false,
			config: {
				latitude: "XXXXX",
				longitude: "XXXXX",
				apikey: "XXXXX"
                                apiVersion: "3.0"
			}
		},
```

Of course you have to replace the XXXXX parts :slightly_smiling_face:

**Comment by KristjanESPERANTO** [comment]
When did you create your apikey?

**Comment by mirrormonark** [comment]
apikey with openweather... apikey only free not buy. 

The OneCall module was a module to take the API is use it more than once but making the call look like only once. That was how it worked.


The OneCall module contained the possibility of having APIs per hour, daily, in anticipation,

**Comment by mirrormonark** [comment]
it doesn't work.. :(

**Comment by KristjanESPERANTO** [comment]
> The OneCall module was a module to take the API is use it more than once but making the call look like only once.

Sounds nice. Where did you get it?

> it doesn't work.. :(

I think we will figure it out. What does `npm run config:check` say? (you have to run in in the MagicMirror folder)

**Comment by KristjanESPERANTO** [comment]
When did you create your apikey? I ask this to find out if it is a key for apiVersion 2.5 or 3.0.

**Comment by mirrormonark** [comment]
npm run config:check say "Your configuration file doesn't contain syntax errors :)"

**Comment by mirrormonark** [comment]
I found this guy who forked the dev... That doesn't work either. Try .
https://github.com/handwrittenjello/onecall

**Comment by mirrormonark** [comment]
maybe you could implement it in your fork

**Comment by mirrormonark** [comment]
I own 2 apikeys. one made this year (2 days ago) while the other was made in 2020 I think so it should be 2.5

**Comment by KristjanESPERANTO** [comment]
> I found this guy who forked the dev...

This is definitely interesting. I'll take a look. I don't have the time for new functions at the moment. But some troubleshooting is okay.

> I own 2 apikeys. one made this year (2 days ago) while the other was made in 2020 I think so it should be 2.5

Please try the following: Remove all modules from your config and add only this module with the new apikey:

```js
		{	
			module: "MMM-OneCallWeather",
			position: "top_right",
			pages: {"daily":"top_right"},
			disabled: false,
			config: {
				latitude: "XXXXX",
				longitude: "XXXXX",
				apikey: "XXXXX",
                                apiVersion: "3.0"
			}
		},
```

If that doesn't work: Which node version (`node -v`) and which npm version (`npm -v`) do you have?

**Comment by mirrormonark** [comment]
I did what you asked (delete all modules) but it always tells me "upload". 
As for the versions here it is:
  node version: v16.19.1
   npm version: 8.19.3

**Comment by mirrormonark** [comment]
hey i see this..  [MMM-OneCallWeather] 14-Mar-23 00:05 ** ERROR ** AxiosError: Request failed with status code 401

**Comment by KristjanESPERANTO** [comment]
Okay, this is an indication that something is wrong with the key.

Can you create a third key and send it to me in a private message via [the forum](https://forum.magicmirror.builders/user/kristjanesperanto)? After the test you can delete it again. I would test how the correct setting must be for your keys.

But I'm going to sleep soon. So I won't answer anymore today.

**Comment by mirrormonark** [comment]
goodnight :)

**Comment by KristjanESPERANTO** [comment]
Something is wrong with the key you sent me. I always get the following error message: `"Invalid API key. Please see https://openweathermap.org/faq#error401 for more info."`  (with the onecall endpoint in version 2.5 and 3.0). But the weather endpoint, on the other hand, works with your key. That's strange.

I also created a new key for myself and it works with that - I have only the Free plan. I'm running out of ideas 😕

**Comment by mirrormonark** [comment]
I apologize, I can ask you all the steps you do on the OpenWeather site... I think to me that I did not enable the OneCall then..

**Comment by KristjanESPERANTO** [comment]
On the [My services page](https://home.openweathermap.org/myservices), the following should be in the "Price Plan" column: "Free Plan".

On the [API Key page](https://home.openweathermap.org/api_keys) you can generate new keys.

**Comment by mirrormonark** [comment]
Good morning. It doesn't work. Can I ask you for a private API to test?

**Comment by mirrormonark** [comment]
Inglese
the weather modules I use are: 
MMM-WEATHERBACKGROUND
MMM-OPENMAPWHEATHER
MMM-NOAA3
MMM-PLACEINFO

I use them on separate pages so not all together. Could it be that one of these gives problems to the module?

**Comment by mirrormonark** [comment]
API calls return an error 401

You can get the error 401 in the following cases:

You did not specify your [API key](https://home.openweathermap.org/api_keys) in API request.
Your API key is not activated yet. Within the next couple of hours, it will be activated and ready to use.
You are using wrong API key in API request. Please, check your right API key in [personal account](https://home.openweathermap.org/api_keys).
You are using a Free subscription and try requesting data available in other subscriptions . For example, [16 days/daily forecast API](https://openweathermap.org/forecast16), [any historical weather data](https://openweathermap.org/price#history), [Weather maps 2.0](https://openweathermap.org/api/weather-map-2), etc). Please, check your subscription in your [personal account](https://home.openweathermap.org/)

**Comment by mirrormonark** [comment]
MMM-weatherforecast this is only work for forecast for me

**Comment by mirrormonark** [comment]
Hourly forecast: unavailable
Daily forecast: unavailable
Calls per minute: 60
3 hour forecast: 5 days

i believe is this the problem. with free api and i believe i have api 2.5 e not 3.0

**Comment by KristjanESPERANTO** [comment]
> Hourly forecast: unavailable
> Daily forecast: unavailable
> Calls per minute: 60
> 3 hour forecast: 5 days

I have the same.

**Comment by mirrormonark** [comment]
but for you working??

**Comment by KristjanESPERANTO** [comment]
Yes, it works perfectly.

**Comment by mirrormonark** [comment]
maybe the problem is because use 4-5 modules with openmap

**Comment by mirrormonark** [comment]
hi how are you? let me inform you that i figured out the problem on this form...i simply paid the site subscription for the plan and everything works. i had not entered any card before.

**Comment by KristjanESPERANTO** [comment]
Interesting. Then new free API keys might not work anymore? 😮

**Comment by mirrormonark** [comment]
i don't think so. simply i did the api onecall plan... i put in my cash card and i have like 1000 free calls if i exceed i pay.

**Comment by Manu85340** [comment]
@KristjanESPERANTO hello I installed your `MMM-OneCallWeather` module thanks to
@sdetweil, he helped me with this module and many other things and I have a question in the readme he wrote there:
"colored If set 'colored' to true the min-temp get a blue tone and the max-temp get a red tone.

Default value: true"
I added this to my js.config but the values remain white am I doing something wrong?

**Comment by KristjanESPERANTO** [comment]
@Manu85340 Please open a new issue and show there your config (without API key) :slightly_smiling_face:

**Comment by KristjanESPERANTO** [comment]
I'm closing this issue because mirrormonark was able to solve the issue by switching to a paid key.
